.gitlab-ci.yml

```diff
@@ -91,6 +91,17 @@ Lint:
     - make clean repo-setup ENV=ci
     - make lint license-check
     - make rum-models-verify sr-models-verify
+    - make api-surface ENV=ci
+    - if ! diff api-surface-swift api-surface-swift-generated; then
+        echo "❌ Swift API surface mismatch";
+        echo "👉 Run \`make api-surface\` locally to update \`api-surface-swift\` and commit the changes.";
+        exit 1;
+      fi
+    - if ! diff api-surface-objc api-surface-objc-generated; then
+        echo "❌ Objective-C API surface mismatch";
+        echo "👉 Run \`make api-surface\` locally to update \`api-surface-objc\` and commit the changes.";
+        exit 1;
+      fi
```

Comment:
suggestion/ Could we move these checks to separate script in `tools/` directory? The convention we started for [.gitlab-ci.yml](https://github.com/DataDog/dd-sdk-ios/blob/develop/.gitlab-ci.yml) is to avoid bare scripts in `yml` as they are very inconvenient to reproduce in local shell.

Ideally, we could get inspired by `make rum-models-verfy`:
* `make api-surface` - to generate API surface files; ran in local
* `make api-surface-verify` - to verify API surface; ran on CI